casper/src/main/scala/coop/rchain/casper/BlockStatus.scala

```diff
@@ -45,6 +45,8 @@ final case object InvalidTransaction      extends InvalidBlock with Slashable
 final case object InvalidBondsCache       extends InvalidBlock with Slashable
 final case object InvalidBlockHash        extends InvalidBlock with Slashable
 final case object InvalidDeployCount      extends InvalidBlock with Slashable
+final case object ContainsExpiredDeploy   extends InvalidBlock with Slashable
+final case object ContainsFutureDeploy    extends InvalidBlock with Slashable
```

Comment:
Does that mean it's a slashable offense for a validator to include a deploy that's already in the log? Wouldn't that be an attack vector? I'm thinking about a situation where an evil validator tries to front-run a good validator by getting a certain deploy in before.

Reply:
Nevermind, I see it now.